gulpfile.js

```diff
@@ -118,7 +117,12 @@ function genproto_group3_commonjs_strict(cb) {
 }
 
 
-function getClosureCompilerCommand(exportsFile, outputFile) {
+function getClosureCompilerCommand(exportsFile, outputFile, keepSymbols) {
```

Comment:
I think one of the main things is to restore the `@protected` (and `@private`, if applicable) annotations. I think the way forward is to let the tests build in simple optimization mode. While they're currently invoking the compiler with `keepSymbols=true`, there's an underlying step that doesn't (`gen_google_protobuf_js`).

Maybe we should fixup the gulp tasks to something like:

```
function enable_advanced_optimizations(cb) {
  USE_ADVANCED_OPTIMIZATIONS = true;
  cb();
}


function disable_advanced_optimizations(cb) {
  USE_ADVANCED_OPTIMIZATIONS = false;
  cb();
}

exports.build = series(exports.build_protoc_plugin,
                      genproto_wellknowntypes,
                      gen_google_protobuf_js);

// ...

exports.test_closure = series(disable_advanced_optimizations, exports.deps,
                              test_closure);

exports.test_commonjs = series(disable_advanced_optimizations, exports.make_commonjs_out,
                               test_commonjs);
// ...

exports.dist = series(enable_advanced_optimizations, exports.build)
```

Reply:
But I think the test should test exactly how the code will be shipped. Exporting those methods will not change the functionality of the code. Maybe we can have an alpha minor version change(3.22.0-alpha) release to see if the bundle size and everything look fine, then release it. `230kB` minified is really too much for this library! and this release should be first priority IMO.

https://bundlephobia.com/package/google-protobuf@3.21.0

Reply:
Ideally, yes, we should be able to test with optimizations off and on. No argument there. Ditto for wanting to cut down the bundle size and get a release out.

My understanding is that it is not uncommon for us to leave the tests unoptimized. While the fully `@export`'d code may be functionally ok, I think it obfuscates the real API and it would be bad if folks started relying on things outside of the contract we want to provide.

@lukesandberg Any thoughts?

Reply:
We really need this, guys. It took more than a month already

Reply:
Luke and I discussed some things offline it seems like, as a library, we are kind of forced to `@export` more than I would otherwise like. So we'll keep going with your current approach and I'll come back later to add some appropriate warning comments to what should be internal stuff.